shortest_path.py

```diff
@@ -1,15 +1,308 @@
+
+        self.start_point = start_point
+        self.target = target
+
+        self.random_respawn_probability = random_respawn_probability
+
+        self.respawn()
+
+        self.pheromone_by_edge = pheromone_by_edge
+        self.pheromone_strength = peheromone_strength
+
+        self.visited_edges = set()
+        self.moves_cost = 0
+
+    def move(self):
+        moves = self.edges_by_vertice[self.position]
+        moves = [i for i in moves if i not in self.visited_edges]
+        if len(moves) == 0:
+            # Going in a loop => die
+            self.respawn()
+            return
+
+        move_probabilities = [1 + self.pheromone_by_edge[i] for i in moves]
+        probabilities_sum = sum(move_probabilities)
+        move_probabilities = [i / probabilities_sum for i in move_probabilities]
+        move = random.choices(moves, move_probabilities)[0]
+
+        self.position = move[0] if move[0] != self.position else move[1]
+        self.visited_edges.add(move)
+        self.moves_cost += self.edges_costs[move] + self.vertices_costs[self.position]
+
+        if self.position == self.target:
+            for edge in self.visited_edges:
+                self.pheromone_by_edge[edge] += self.pheromone_strength / math.log(self.moves_cost)
+
+            self.respawn()
+
+
+    def respawn(self):
+        self.position = random.choice([i for i in self.vertices]) if random.randrange(int(100 / self.random_respawn_probability)) < 100 else self.start_point
+        self.visited_edges = set()
+        self.moves_cost = 0
+
 
 def find_shortest_path(img, x1, y1, x2, y2):
     sleep(1)
-    print(type(x1), x2, y1, y2)
+    X = np.linspace(0, 1, 64)
+    Y = np.linspace(0, 1, 64)
+    r = 1.5
+    v1 = sum([sum([1 for x in X if (x-r)**2+(y-r)**2 <= r*r]) for y in Y]) / 64**2
+    v2 = sum([sum([1 for x in X if (x-r)**2+y**2 <= r*r]) for y in Y]) / 64**2
+
+
+    mask = np.array([[v1, v2, v1], [v2, 1.0, v2], [v1, v2, v1]])
+
+    for i in range(4):
+        mask = mask * mask
+        s = sum(sum(mask)) - mask[1, 1]
+        mask[1, 1] = s
+        mask = mask / s
+
+    mask = -mask
+    mask[1, 1] = -mask[1, 1]
+
+    gray = img[::, ::, 0]
+    gray = cv2.GaussianBlur(gray,(5,5), 0)
+    gray01 = gray.astype(float) / 255
+    dst = gray.astype(float) / 255
+
+    dst = cv2.filter2D(dst, -1, mask)
+    _min = dst.min()
+    _max = dst.max()
+    dst = (dst-_min)/(_max-_min)
+
+    dst = dst**2
+
+#################################################################
+    input_image = prog(dst*255, 55)
+    start_point = (int(x1), int(y1))
+    target = (int(x2), int(y2))
+    reachable_image = get_reachable_image(input_image, start_point)
+    weighted_image = get_weighted_image(reachable_image)
+    skeletonized_image = get_skeletonized_image(reachable_image,weighted_image)
+    reachable_points = get_reachable_points(skeletonized_image)
+    
+    intersection_points = get_intersection_points(reachable_points, start_point, target)
+
+
```

Comment:
Remooooooovvvveeeeee plllleeeeaaaasssseeeee eemmmmmmppptyyyyyyyy liiiinnnneeeesssss

Reply:
or use Python format in VS Code